File: app/src/main/java/java/android/notes/wrapper/CalendarFragment.java
package java.android.notes.wrapper;

import androidx.fragment.app.Fragment;

import android.app.DatePickerDialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.DatePicker;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.android.notes.R;
import java.android.notes.Time;

public class CalendarFragment extends Fragment{

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        return inflater.inflate(R.layout.fragment_calendar,container,false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextView textViewDate = view.findViewById(R.id.textViewDate);
        textViewDate.setText(Time.getDateYMD());

        ////////////////////////////////////////////////////////////////
        DatePickerDialog.OnDateSetListener setListener = new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker datePicker, int i, int i1, int i2) {
                Time.setDay(i2);
                Time.setMonth(i1);
                Time.setYear(i);

                textViewDate.setText(Time.getDateYMD());
            }
        };

        DatePickerDialog datePickerDialog = new DatePickerDialog(getContext(),setListener,Time.getYear(),Time.getMonth(),Time.getDay());
        ///////////////////////////////////////////////////////////////

        view.findViewById(R.id.buttonEditDate).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                datePickerDialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
                datePickerDialog.show();
            }
        });

    }

}

File: app/src/main/java/java/android/notes/wrapper/fragments/extra/CalendarFragment.kt
package java.android.notes.wrapper.fragments.extra

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import java.android.notes.R
import android.widget.TextView
import android.app.DatePickerDialog.OnDateSetListener
import android.widget.DatePicker
import android.app.DatePickerDialog
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.View
import androidx.fragment.app.Fragment
import java.android.notes.Time

class CalendarFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.fragment_calendar, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val textViewDate = view.findViewById<TextView>(R.id.textViewDate)
        textViewDate.text = Time.dateYMD

        ////////////////////////////////////////////////////////////////
        val setListener = OnDateSetListener { datePicker, i, i1, i2 ->
            Time.day = i2
            Time.month = i1
            Time.year = i
            textViewDate.text = Time.dateYMD
        }
        val datePickerDialog =
            DatePickerDialog(context!!, setListener, Time.year, Time.month, Time.day)
        ///////////////////////////////////////////////////////////////
        view.findViewById<View>(R.id.buttonEditDate).setOnClickListener {
            datePickerDialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            datePickerDialog.show()
        }
    }
}